anchore_engine/clients/grype_wrapper.py

```diff
@@ -50,6 +50,12 @@ def deserialized_related_vulnerabilities(self):
     def deserialized_fixed_in_versions(self):
         return json.loads(self.fixed_in_versions)
 
+    @property
+    def deserialized_version_constraint(self):
+        return [
```

Comment:
version_constraint could be empty/blank. It means the vulnerability has not been fixed implying all versions of the package are vulnerable (for the matching process). We see this often in both os and non-os packages. For visual representation [legacy mode](https://github.com/anchore/anchore-engine/blob/master/anchore_engine/services/policy_engine/engine/vulns/providers.py#L744) uses wild character `*`.   
I would recommend returning the version_constraint string as it is here and moving the parsing/interpreting logic to the usage. That could check if the version_constraint is blank and use a wild char  or parse the versions and populate them accordingly 

```
sqlite> SELECT count(*) from vulnerability where version_constraint='';
count(*)
82354

sqlite> SELECT namespace, count(*) from vulnerability where version_constraint='' group by namespace order by namespace;
namespace|count(*)
debian:10|3084
debian:11|1513
debian:12|1387
debian:9|4754
debian:unstable|1558
msrc:10379|56
msrc:10543|44
msrc:10729|59
msrc:10735|60
msrc:10788|49
msrc:10789|49
msrc:10816|58
msrc:10852|56
msrc:10853|57
msrc:10855|59
msrc:10951|16
msrc:10952|25
msrc:11453|14
msrc:11454|14
msrc:11466|23
msrc:11497|32
msrc:11498|35
msrc:11499|22
msrc:11563|17
msrc:11568|34
msrc:11569|36
msrc:11570|16
msrc:11571|31
msrc:11572|31
msrc:11583|14
msrc:11644|40
msrc:11645|42
msrc:11646|40
msrc:11647|36
msrc:11712|14
msrc:11713|16
msrc:11714|15
msrc:11715|8
msrc:11766|105
msrc:11767|104
msrc:11768|105
msrc:11769|100
msrc:products|37
nvd|13088
rhel:5|6195
rhel:6|4462
rhel:7|4360
rhel:8|1710
ubuntu:12.04|3351
ubuntu:12.10|1320
ubuntu:13.04|1023
ubuntu:14.04|3931
ubuntu:14.10|2207
ubuntu:15.04|2392
ubuntu:15.10|3005
ubuntu:16.04|737
ubuntu:16.10|2461
ubuntu:17.04|2643
ubuntu:17.10|2048
ubuntu:18.04|3860
ubuntu:18.10|1485
ubuntu:19.04|2311
ubuntu:19.10|1559
ubuntu:20.04|1857
ubuntu:20.10|1289
ubuntu:21.04|1195
```
```

Reply:
Hey swathi, the empty case is already handled by Dan's code in mappers.py line 890. If you want me to move things around I can, but the edge case is already covered.

Reply:
Would be good to add null check to protect from unexpected db changes. Just to be clear, missing fix version is *not* an edge case. There are over 80K vuln records in the database

Reply:
As requested, I've moved all of the parsing to the caller (EngineGrypeDBMapper.to_engine_vulnerabilities()). I've also identified another case where affected version is "Unspecified". Also i've included some reusable dedup logic for the affected packages.